src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```diff
@@ -346,14 +349,51 @@ public ProjectRepo getUpdatedRepo(
      */
     private ProjectRepo getUpdatedRepoCritical(
             Optional<Credential> oauth2,
-            String projectName
+            String projectName,
+            GetDocResult doc
     ) throws IOException, GitUserException {
         ProjectRepo repo;
         ProjectState state = dbStore.getProjectState(projectName);
         switch (state) {
         case NOT_PRESENT:
-            repo = repoStore.initRepo(projectName);
-            break;
+            Log.info("[{}] Repo not present", projectName);
+            String migratedFromID = doc.getMigratedFromID();
+            if (migratedFromID != null) {
+                Log.info("[{}] Has a migratedFromId: {}", projectName, migratedFromID);
+                ProjectState sourceState = dbStore.getProjectState(migratedFromID);
```

Comment:
Is there a reason to check the state outside the lock? It seem like it might create a race.

Reply:
Yeah, inside the lock would be better.